test/common_quantized.py

```diff
@@ -34,3 +36,24 @@ def _requantize(x, multiplier, zero_point, qmin=0, qmax=255, qtype=np.uint8):
     qx = (x * multiplier).round() + zero_point
     qx = np.clip(qx, qmin, qmax).astype(qtype)
     return qx
+
+def _calculate_dynamic_qparams(X, dtype):
+    """Calculate the dynamic quantization parameters (scale, zero_point)
+    according to the min and max element of the tensor"""
+    if dtype == torch.qint8:
+        qmin, qmax = -128, 127
+    else:  # dtype == torch.quint8
+        qmin, qmax = 0, 255
+    n_levels = 255.0
+    min_val = torch.min(X).item()
+    max_val = torch.max(X).item()
+    if min_val == max_val:
+        scale = 1.0
+        zero_point = 0
+    else:
+        scale = (max_val - min_val) / n_levels
```

Comment:
Should be:
```
max_val = torch.max(max_val, torch.tensor([0.0])
min_val = torch.min(min_val,torch.tensor([0.0])
```
 I dont see the code for fbGemm::choosequantizationparams or findminmax, but we need to ensure that this logic is implemented there too.

Reply:
Why do we need to compare the `max_val` and `min_val` with 0.0?

I followed the same logic in the observer for calculating the quantization parameters:
https://github.com/pytorch/pytorch/blob/master/torch/quantization/observer.py#L39-L64

Reply:
Oh, I know why we need to compare with 0 here: We extend the [min, max] interval to ensure that it contains 0. Otherwise, we would not meet the requirement that 0 be an exactly representable value.

The logic for fbgemm::choosequantizationparams is here:
https://github.com/pytorch/FBGEMM/blob/master/src/QuantUtils.cc#L43-L47

Reply:
Updated the code.

Reply:
The observer code has an error that I am fixing, wanted to make sure it doesnt propagate.